store/src/java/com/zimbra/cs/service/admin/ModifyAccount.java

```diff
@@ -227,6 +227,22 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
         try {
             // pass in true to checkImmutable
             prov.modifyAttrs(account, attrs, true);
+
+            // If request is for `zimbraTwoFactorAuthEnabled = FALSE` and both zimbraFeatureTwoFactorAuthAvailable and zimbraFeatureTwoFactorAuthRequired is set to TRUE
+            // for the user, clear all the trusted devices so that the user would re-setup 2FA in the next login after LDAP updation is successful.
+            if (attrs.containsKey(Provisioning.A_zimbraTwoFactorAuthEnabled)) {
+                String value = (String) attrs.get(Provisioning.A_zimbraTwoFactorAuthEnabled);
+                boolean is2faAvailable = account.isFeatureTwoFactorAuthAvailable();
```

Comment:
Suggestion: account.isFeatureTwoFactorAuthAvailable() and account.isFeatureTwoFactorAuthRequired() could be moved as if conditions directly.
Reason: In current programmed scenario variable assignment will still happen in even those scenario where the following `if` condition fails in either of the first two checks.

Reply:
thanks. will remove the variable assignmenmt.